include/path_tracking_pid/path_tracking_pid_local_planner.hpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include "./controller.hpp"
+#include <atomic>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/thread/recursive_mutex.hpp>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <costmap_2d/costmap_2d_ros.h>
+#include <dynamic_reconfigure/server.h>
+#include <mbf_costmap_core/costmap_controller.h>
+#include <mbf_msgs/ExePathResult.h>
+#include <nav_core/base_local_planner.h>
+#include <nav_msgs/Odometry.h>
+#include <nav_msgs/Path.h>
+#include <path_tracking_pid/PidConfig.h>
+#include <path_tracking_pid/PidFeedback.h>
+#include <ros/ros.h>
+#include <std_msgs/Bool.h>
+#include <std_msgs/Float64.h>
+#include <tf/transform_listener.h>
+#include <tf2_ros/buffer.h>
+#include <visualization_msgs/Marker.h>
+#include <visualization_msgs/MarkerArray.h>
+
+#define MAP_PARALLEL_THRESH 0.2
+constexpr double DT_MAX=1.5;
+
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(geometry_msgs::Point, double, cs::cartesian, x, y)
+
+namespace path_tracking_pid
+{
+class TrackingPidLocalPlanner : public nav_core::BaseLocalPlanner, public mbf_costmap_core::CostmapController
+{
+private:
+  typedef boost::geometry::model::ring<geometry_msgs::Point> polygon_t;
+
+  inline polygon_t union_(polygon_t polygon1, polygon_t polygon2)
+  {
+    std::vector<polygon_t> output_vec;
+    boost::geometry::union_(polygon1, polygon2, output_vec);
+    return output_vec.at(0);  // Only first vector element is filled
```

Comment:
The union of two polygons can be two polygons if there is no overlap. So this funciton always throws away the second

Reply:
Ah, good point. We expect them to overlap though, so we should probably warn on size > 1